standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -179,90 +93,12 @@ public Set<CompactionInfo> findPotentialCompactions(int abortedThreshold, long a
-          "WHERE \"C\".\"CC_ID\" IS NOT NULL OR " + isWithinCheckInterval("\"TC\".\"CTC_TIMESTAMP\"", checkInterval) : ""));
-
-        LOG.debug("Going to execute query <{}>", s);
-
-        rs = stmt.executeQuery(s);
-        while (rs.next()) {
-          CompactionInfo info = new CompactionInfo();
-          info.dbname = rs.getString(1);
-          info.tableName = rs.getString(2);
-          info.partName = rs.getString(3);
-          response.add(info);
-        }
-        rs.close();
-
-        if (!MetastoreConf.getBoolVar(conf, ConfVars.COMPACTOR_CLEAN_ABORTS_USING_CLEANER)) {
-          // Check for aborted txns: number of aborted txns past threshold and age of aborted txns
-          // past time threshold
-          boolean checkAbortedTimeThreshold = abortedTimeThreshold >= 0;
-          String sCheckAborted = "SELECT \"TC_DATABASE\", \"TC_TABLE\", \"TC_PARTITION\", " +
-                  "MIN(\"TXN_STARTED\"), COUNT(*) FROM \"TXNS\", \"TXN_COMPONENTS\" " +
-                  "   WHERE \"TXN_ID\" = \"TC_TXNID\" AND \"TXN_STATE\" = " + TxnStatus.ABORTED + " " +
-                  "GROUP BY \"TC_DATABASE\", \"TC_TABLE\", \"TC_PARTITION\" " +
-                  (checkAbortedTimeThreshold ? "" : " HAVING COUNT(*) > " + abortedThreshold);
-
-          LOG.debug("Going to execute query <{}>", sCheckAborted);
-          rs = stmt.executeQuery(sCheckAborted);
-          long systemTime = System.currentTimeMillis();
-          while (rs.next()) {
-            boolean pastTimeThreshold =
-                    checkAbortedTimeThreshold && rs.getLong(4) + abortedTimeThreshold < systemTime;
-            int numAbortedTxns = rs.getInt(5);
-            if (numAbortedTxns > abortedThreshold || pastTimeThreshold) {
-              CompactionInfo info = new CompactionInfo();
-              info.dbname = rs.getString(1);
-              info.tableName = rs.getString(2);
-              info.partName = rs.getString(3);
-              info.tooManyAborts = numAbortedTxns > abortedThreshold;
-              info.hasOldAbort = pastTimeThreshold;
-              LOG.debug("Found potential compaction: {}", info);
-              response.add(info);
-            }
-          }
-        }
-      } catch (SQLException e) {
-        LOG.error(DB_FAILED_TO_CONNECT + e.getMessage());
-        checkRetryable(e,
-            "findPotentialCompactions(maxAborted:" + abortedThreshold
-                + ", abortedTimeThreshold:" + abortedTimeThreshold + ")");
-      } finally {
-        close(rs, stmt, dbConn);
-      }
-      return response;
-    }
-    catch (RetryException e) {
-      return findPotentialCompactions(abortedThreshold, abortedTimeThreshold, lastChecked);
+    Set<CompactionInfo> candidates = new HashSet<>(jdbcTemplate.execute(
+        new CompactionCandidateHandler(lastChecked, MetastoreConf.getIntVar(conf, ConfVars.COMPACTOR_FETCH_SIZE))));
+    if (!MetastoreConf.getBoolVar(conf, ConfVars.COMPACTOR_CLEAN_ABORTS_USING_CLEANER)) {
```

Comment:
since we introduced wrappers, can't we move complete logic there (handle aborts)?